File: cpp/RT_CPP/RT_CPP/texture.cpp
#include "texture.h"


using namespace utility;

texture::texture(vector3** pixels, int width, int height) : pixels(pixels), width(width), height(height)
{

}

texture::~texture()
{
	for (int i = 0; i < height; i++)
	{
		delete pixels[i];
	}
	delete pixels;
}

vector3 texture::get_pixel(vector2 uv)
{
	int x = (int)fmaxf(0, roundf(uv.x * width) - 1);
	int y = (int)fmaxf(0, roundf(uv.y * height) - 1);
	return pixels[y][x];
}

File: Csharp/RT_CS/RT_CS/utility/Texture.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace RT_CS.utility
{
    public class Texture 
    {
        private Vector3[,] pixels;
        private int width, height;

        public Texture(Vector3[,] pxs, int w, int h)
        {
            pixels = pxs;
            width = w;
            height = h;
        }

        public Vector3 GetPixel(Vector2 uv)
        {
            int x = (int)Math.Max(0, Math.Round(uv.x * width) - 1);
            int y = (int)Math.Max(0, Math.Round(uv.y * height) - 1);
            return pixels[y,x];

        }
    }
}
